File: app/src/main/java/com/phucnguyen/lovereminder/ui/fragment/dialog/DialogFragment.java
package com.phucnguyen.lovereminder.ui.fragment.dialog;

import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.EditText;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatDialogFragment;

import com.phucnguyen.lovereminder.R;

public class DialogFragment extends AppCompatDialogFragment {
    private EditText editTextUsername;
    private Listener listener;
    private static final String TAG = "MyCustomDialog";
    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        try{
            listener = (Listener) getTargetFragment();
        }catch (ClassCastException e){
            Log.e(TAG, "onAttach: ClassCastException : " + e.getMessage() );
        }

    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());

        LayoutInflater inflater = getActivity().getLayoutInflater();
        View view = inflater.inflate(R.layout.layout_dialog, null);

        builder.setView(view)
                .setTitle("Sửa Tên")
                .setNegativeButton("Hủy", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {

                    }
                })
                .setPositiveButton("Lưu", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        String username = editTextUsername.getText().toString();
                        listener.ApplyChange(username);
                    }
                });

        String name = getArguments().getString("name");
        editTextUsername = view.findViewById(R.id.edt_userName);
        editTextUsername.setText(name);



        return builder.create();
    }
    public interface Listener{
        void ApplyChange(String username);
    }
}

File: app/src/main/java/com/phucnguyen/lovereminder/ui/fragment/dialog/DialogFragment.kt
package com.phucnguyen.lovereminder.ui.fragment.dialog

import android.app.Dialog
import android.content.Context
import androidx.appcompat.app.AppCompatDialogFragment
import android.widget.EditText
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import androidx.appcompat.app.AlertDialog
import com.phucnguyen.lovereminder.R
import java.lang.ClassCastException

class DialogFragment : AppCompatDialogFragment() {
    private var editTextUsername: EditText? = null
    private var listener: Listener? = null
    override fun onAttach(context: Context) {
        super.onAttach(context)
        try {
            listener = targetFragment as Listener?
        } catch (e: ClassCastException) {
            Log.e(TAG, "onAttach: ClassCastException : " + e.message)
        }
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(
            activity!!
        )
        val inflater = activity!!.layoutInflater
        val view = inflater.inflate(R.layout.layout_dialog, null)
        builder.setView(view)
            .setTitle("Sửa Tên")
            .setNegativeButton("Hủy") { dialogInterface, i -> }
            .setPositiveButton("Lưu") { dialogInterface, i ->
                val username = editTextUsername!!.text.toString()
                listener!!.ApplyChange(username)
            }
        val name = arguments!!.getString("name")
        editTextUsername = view.findViewById(R.id.edt_userName)
        editTextUsername?.setText(name)
        return builder.create()
    }

    interface Listener {
        fun ApplyChange(username: String?)
    }

    companion object {
        private const val TAG = "MyCustomDialog"
    }
}